File: backend/Pedido/producer.py
import pika
import json
import os

from dotenv import load_dotenv

load_dotenv()


def enviar_mensaje(mensaje):

    try:

        rabbit_url = os.getenv("RABBIT_URL")

        print("Conectando Rabbit:", rabbit_url)

        if not rabbit_url:

            print("RABBIT_URL no está configurada")
            return

        parametros = pika.URLParameters(rabbit_url)

        conexion = pika.BlockingConnection(parametros)

        canal = conexion.channel()

        canal.queue_declare(
            queue="pedidos",
            durable=True
        )

        canal.basic_publish(
            exchange="",
            routing_key="pedidos",
            body=json.dumps(mensaje),
            properties=pika.BasicProperties(
                delivery_mode=2
            )
        )

        print("Mensaje enviado a cola pedidos")

        conexion.close()

    except Exception as e:

        print("RabbitMQ no disponible")
        print(e)

File: backend/Inventario/producer.js
const amqp = require("amqplib");

async function enviarMensaje(mensaje) {

    try {

        console.log("Conectando Rabbit:", process.env.RABBIT_URL);

        const conexion = await amqp.connect(
            process.env.RABBIT_URL
        );

        const canal = await conexion.createChannel();

        await canal.assertQueue(
            "inventario",
            {
                durable: true
            }
        );

        canal.sendToQueue(
            "inventario",
            Buffer.from(JSON.stringify(mensaje)),
            {
                persistent: true
            }
        );

        console.log("Mensaje enviado a cola inventario");

        await canal.close();
        await conexion.close();

    } catch (error) {

        console.log("RabbitMQ no disponible");
        console.log(error.message);
    }
}

module.exports = enviarMensaje;